cmd/itzo-launcher/itzo-launcher.go

```diff
@@ -70,7 +70,11 @@ func ProcessInstanceParameters() error {
 
 func ProcessUserData() error {
 	klog.V(2).Infof("getting itzo files from cloud-init")
-	err := cloudinit.WriteFiles(ItzoURLFile, ItzoVersionFile, CellConfigFile)
+	err := os.MkdirAll(ItzoDir, os.ModeDir)
```

Comment:
Since we might want to write other files too from here, it might be better to move the directory creation into `cloudinit.WriteFiles`. You can check the directory of the file to be written there and ensure it exists before writing the file.

Reply:
Okay I have moved the Mkdir inside of `cloudinit.WriteFiles` right before we start iterating over the files to create. In order to do this we have to also pass the desired directory to create to `cloudinit.WriteFiles` so the signature now looks as follows `cloudinit.WriteFiles(dir string, paths ...string)`.